test/Web3Task.test.ts

```diff
@@ -3,386 +3,868 @@ import { Contract, ContractFactory } from "ethers";
+    ).to.be.revertedWithCustomError(Web3Task, "Unauthorized");
+    const currentQuorum = await Web3Task.APPROVALS();
+    expect(currentQuorum).to.equal(initialQuorum);
+  });
+
+  //New-test - Testing setting minQuorum with zero or negative value
+  it("should fail when a negative value is set for the minimum quorum", async function () {
+    await expect(Web3Task.connect(owner).setMinQuorum(-2)).to.be.rejected;
+  });
+
+  //New-test - Testing creating a task with an invalid status (1 - 4)
+  it("should falied to create new task ( invalid status - Progress (1), Review (2), Completed (3), Canceled (4) )", async function () {
+    const invalidStatuses = [1, 2, 3, 4];
+    for (const status of invalidStatuses) {
+      const Task = {
+        status: status,
+        title: "Invalid status task",
+        description: "Não esquecer",
+        reward: ethers.utils.parseEther("1"),
+        endDate: ethers.constants.MaxUint256,
+        authorizedRoles: [memberId],
+        creatorRole: leaderId,
+        assignee: userC.address,
+        metadata: "ipfs://0xc0/",
+      };
+      await expect(
+        Web3Task.connect(userA).createTask(Task)
+      ).to.be.revertedWithCustomError(Web3Task, "InvalidStatus");
+    }
+  });
+
+  //New-test - Testing creating two tasks with the same parameters to return different IDs
+  it("should create a new task with the same parameters as different tasks, not duplicate", async function () {
+    const Task = {
+      status: 0,
+      title: "Task with same parameters",
+      description: "Task description",
+      reward: ethers.utils.parseEther("1"),
+      endDate: ethers.constants.MaxUint256,
+      authorizedRoles: [memberId],
+      creatorRole: leaderId,
+      assignee: userC.address,
+      metadata: "ipfs://0xc0/",
+    };
+
+    let tx = await Web3Task.connect(userA).createTask(Task);
+    let receipt = await tx.wait();
+    let firstTaskId = receipt.events[0].args[0];
+
+    tx = await Web3Task.connect(userA).createTask(Task);
+    receipt = await tx.wait();
+    let secondTaskId = receipt.events[0].args[0];
+
+    expect(firstTaskId).to.not.equal(secondTaskId);
+  });
+
+  //New-test - Testing creating a task with an invalid reward (0)
+  it("should fail to create a new task (invalid reward - 0)", async function () {
+    const Task = {
+      status: 0,
+      title: "invalid reward",
+      description: "Não esquecer",
+      reward: ethers.utils.parseEther("0"),
+      endDate: ethers.constants.MaxUint256,
+      authorizedRoles: [memberId],
+      creatorRole: leaderId,
+      assignee: userB.address,
+      metadata: "ipfs://0xc0/",
+    };
+    await expect(
+      Web3Task.connect(userA).createTask(Task)
+    ).to.be.revertedWithCustomError(Web3Task, "InvalidReward");
+  });
+
+  //New-test - Testing creating a task with an invalid reward (negative value)
+  it("should fail to create new task (invalid reward - negative value)", async function () {
```

Comment:
rewards cannot be negative by default as they use uint256 which is only made of positive integers